File: task-manager-frontend/src/services/taskService.js
import api from "./api";

const taskService = {
    
    getAllTasks: async () => {
        const response = await api.get("/tasks");
        return response.data;
    },

    getTaskById: async (id) => {
        const response = await api.get(`/tasks/${id}`);
        return response.data;
    },

    createTask: async (taskData) => {
        const response = await api.post("/tasks", taskData);
        return response.data;
    },

    updateTask: async (id, taskData) => {
        const response = await api.put(`/tasks/${id}`, taskData);
        return response.data;
    },

    deleteTask: async (id) => {
        const response = await api.delete(`/tasks/${id}`);
        return response.data;
    },

    updateTaskStatus: async (id, status) => {
        const response = await api.patch(`/tasks/${id}/status?status=${status}`);
        return response.data;
    }
};

export default taskService;

File: backend/src/main/java/com/taskmanager/service/TaskService.java
package com.taskmanager.service;

import java.util.List;

import com.taskmanager.dto.request.TaskRequest;
import com.taskmanager.dto.response.TaskResponse;

public interface TaskService {

    TaskResponse createTask(TaskRequest request);

    List<TaskResponse> getAllTasks();

    TaskResponse getTaskById(Long id);

    TaskResponse updateTask(Long id, TaskRequest request);

    void deleteTask(Long id);

    TaskResponse updateTaskStatus(Long id, String status);
}